### ISSUE TITLE
Allow interaction with smart contract in offline mode by passing nonce

### ISSUE THREAD

**Issue by rstormsf** [MyCryptoHQ/MyCrypto#1495]
Right now there is no way to pregenerate a signed tx in offline mode because Contracts tab uses getTransactionCount to get nonce

It would be great if a user can manually insert it

**Comment by wbobeirne** [comment]
You're absolutely right, in fact you can't even access the tab while offline (It just shows the offline view.) There's no reason you shouldn't be able to generate a signed transaction on the contracts tab while offline.

It'll probably take a bit of refactoring to get this tab working, so what I'd say is that in the mean time, you'll probably want to:
* Use a bogus private key (not MetaMask) and generate the transaction with the inputs you want
* Copy the `data` field from the signed transaction
* Go offline, unlock the wallet you actually want to send from on the send view
* Toggle advanced mode and paste the data in that you copied from before
* Generate transaction
* Broadcast signed transaction when online

This is definitely not the most fun process, but it should work without having to sign online.

**Comment by dec1** [comment]
> Copy the data field from the _signed transaction_

I take it you mean from the **raw transaction**?

> Go offline, ....

Any use the contract address as the "To Address" ?

**Comment by wbobeirne** [comment]
Correct on both accounts, sorry for the mistake, edited post to fix.

**Comment by gernotpokorny** [comment]
Is it supported in the meantime?

**Comment by wbobeirne** [comment]
@gernotpokorny The instructions in https://github.com/MyCryptoHQ/MyCrypto/issues/1495#issuecomment-381275053 do work at the moment on beta.mycrypto.com

**Comment by dec1** [comment]
Is there some similar way to add  a custom token and sign a (send) transaction (offline) and then send online?

**Comment by Azarielle** [comment]
[Added to Asana](https://app.asana.com/0/763966770926912/779983315837085)
